doc/src/SUMMARY.md

```diff
@@ -18,3 +17,8 @@
 - [Equivalence checking](./equivalence.md)
 - [Concrete execution](./exec.md)
 - [Common options](./common-options.md)
+
+# Development
+
+- [Developers tutorial](./for-hevm-developers.md)
+- [System architecture](./architecture.md)
```

Comment:
Moved development into its own sub-section, with the `architecture.md` moved here, so it's easier to access.